src/components/StudioTopbar/StudioTopbar.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { Text, Icon } from '@/shared/components'
+import {
+  StyledTopbarBase,
+  StudioContainer,
+  DrawerButton,
+  AddVideoButton,
+  MemberInfoContainer,
+  MemberTitleText,
+  MemberTextContainer,
+  ChannelInfoContainer,
+  StyledAvatar,
+  TextContainer,
+  DrawerContainer,
+  NewChannel,
+  NewChannelIconContainer,
+  StyledLink,
+  LogoutButton,
+} from './StudioTopbar.style'
+
+type Channel = {
+  name: string
+  avatar: string
+}
+type ChannelInfoProps = {
+  active?: boolean
+  member: string
+  channel: Channel
+  onClick?: React.MouseEventHandler<HTMLDivElement>
+}
+
+type MemberInfoProps = {
+  memberName: string
+  memberAvatar: string
+}
+
+type NavDrawerProps = {
+  active?: boolean
+  memberName: string
+  memberAvatar: string
+  channels: Channel[]
+  currentChannel: Channel
+  onCurrentChannelChange: (channel: Channel) => void
+  onLogoutClick: () => void
+  handleClose: () => void
+}
+
+const member = {
+  name: 'Mikael Cowan',
+  avatar: 'https://picsum.photos/300/300',
+  channels: [
+    { name: 'Wild Crypto Fan16', avatar: 'https://picsum.photos/201/300' },
+    { name: 'Mild Crypto Skeptic', avatar: 'https://picsum.photos/202/300' },
+    { name: 'Average Cryptocurrency Enjoyer', avatar: 'https://picsum.photos/203/300' },
+  ],
+}
+
+const StudioTopbar: React.FC = () => {
+  const [isDrawerActive, setDrawerActive] = useState(false)
+  // TODO Change that to use hook for saving/getting currently active channel
```

Comment:
I guess this can be integrated now?

Reply:
Yup, but we still need membership queries to make it work